### PR TITLE
Add cohttp<0.22.0, lwt<3.0.0, mirage<3.0.0 bounds to all our packages

### PR DESCRIPTION
**Pull request by gaborigloi**
And fix some outdated/out-of-sync packages.

### CODE REVIEW

**Comment by gaborigloi** [comment]
Done with `sed`:
```
sed -i 's/"cohttp"$/"cohttp" {< "0.22.0"}/' packages/xs-extra/*/opam
sed -i 's/"cohttp"$/"cohttp" {< "0.22.0"}/' packages/xs/*/opam
sed -i 's/"cohttp" {>= \(.*\)}$/"cohttp" {>= \1 \& < "0.22.0"}/' packages/xs/*/opam
sed -i 's/"cohttp" {>= \(.*\)}$/"cohttp" {>= \1 \& < "0.22.0"}/' packages/xs-extra/*/opam

sed -i 's/"lwt"$/"lwt" {< "3.0.0"}/' packages/xs-extra/*/opam
sed -i 's/"lwt"$/"lwt" {< "3.0.0"}/' packages/xs/*/opam
sed -i 's/"lwt" {>\(.*\)}$/"lwt" {>\1 \& < "3.0.0"}/' packages/xs-extra/*/opam
sed -i 's/"lwt" {>\(.*\)}$/"lwt" {>\1 \& < "3.0.0"}/' packages/xs/*/opam
...
```

**Comment by gaborigloi** [comment]
Btw, can we now remove the `lwt` pinning step (it's pinned to 2.7.1) from the `Dockerfile`, README, and `dev-vm`? Now we have `lwt < 3.0.0` for all our packages.

File: packages/xs-extra/xapi-nbd.master/opam [hunk 1]
```diff
@@ -24,7 +24,7 @@ depends: [
   "sexplib"
   "mirage-block-unix"
   "io-page"
-  "mirage"
+  "mirage" {< "3.0.0"}
```
Comment (mseri) [inline packages/xs-extra/xapi-nbd.master/opam]: Please use https://github.com/ocaml/opam-repository/blob/master/packages/nbd/nbd.2.1.3/opam for nbd, I forgot to push the update to xs-opam

File: packages/xs/nbd.2.1.3/opam [hunk 2]
```diff
@@ -27,13 +27,13 @@ depends: [
...
   "ppx_tools" {build}
-  "lwt"
+  "lwt" {< "3.0.0"}
   "cstruct" {>= "1.9.0" & < "3.0.0"}
   "cmdliner"
   "sexplib"
   "mirage-block-unix"
   "io-page"
-  "mirage" {>= "1.1.0"}
+  "mirage" {>= "1.1.0" & < "3.0.0"}
```
Comment (mseri) [inline packages/xs/nbd.2.1.3/opam]: Please use https://github.com/ocaml/opam-repository/blob/master/packages/nbd/nbd.2.1.3/opam for nbd, I forgot to push the update to xs-opam

File: packages/xs-extra/xapi-nbd.master/opam (hunk 1 above)
Comment (gaborigloi) [inline packages/xs-extra/xapi-nbd.master/opam]: Don't think so, we should just remove the dependencies. I think they are only transitive deps.

Comment (gaborigloi) [inline packages/xs-extra/xapi-nbd.master/opam]: Thanks for the suggestion, cleaned up the `opam` file, now it works with both `xs-opam` and the upstream repo: https://travis-ci.org/gaborigloi/xapi-nbd/builds/237973155

**Comment by mseri** [comment]
We still have an error that I don't like:

```
#=== ERROR while installing xapi-rrd-transport.master =========================#
# opam-version 1.2.2
# os           linux
# command      make
# path         /home/travis/.opam/system/build/xapi-rrd-transport.master
# compiler     system (4.02.3)
# exit-code    2
# env-file     /home/travis/.opam/system/build/xapi-rrd-transport.master/xapi-rrd-transport-2023-809718.env
# stdout-file  /home/travis/.opam/system/build/xapi-rrd-transport.master/xapi-rrd-transport-2023-809718.out
# stderr-file  /home/travis/.opam/system/build/xapi-rrd-transport.master/xapi-rrd-transport-2023-809718.err
### stdout ###
# ocaml setup.ml -configure --enable-tests
### stderr ###
# ocamlfind: Package `xen-gnt.unix' not found
# W: Field 'pkg_xen_gnt_unix' is not set: Command ''/home/travis/.opam/system/bin/ocamlfind' query -format %d xen-gnt.unix > '/tmp/oasis-b53d24.txt'' terminated with error code 2
# E: Cannot find findlib package xen-gnt.unix
# E: Failure("1 configuration error")
# make: *** [setup.data] Error 1
```

**Comment by gaborigloi** [comment]
Yeah, we need to add `xen-gnt < 3.0.0`, it has been split into two OPAM packages. From its [changelog](https://github.com/mirage/ocaml-gnt/blob/master/CHANGES.md):
## 3.0.0 (2017-05-28):
* Split into 2 opam and ocamlfind packages:
  xen-gnt: same as before
  xen-gnt-unix: this used to be called xen-gnt.unix
* Add dependencies for archlinux
* Switch to jbuilder

**Comment by mseri** [comment]
:+1:

**Comment by gaborigloi** [comment]
PR for adding `xen-gnt` bounds: https://github.com/xapi-project/xs-opam/pull/94

**Pull request merged**